[DOCS] Adds 8.11.0 release notes (#168593)
elastic/kibana: docs/CHANGELOG.asciidoc

.Improve config output validation for default output.
[%collapsible]
====
*Details* +
Improve config output validation to not allow to defining multiple default outputs in {kib} configuration. For more information, refer to ({kibana-pull}167085[#167085]).
====
      
[discrete]
[[breaking-161806]]
.Convert filterQuery to KQL.
[%collapsible]
====
*Details* +
Converts `filterQuery` to a KQL query string. For more information, refer to ({kibana-pull}161806[#161806]).
====
      
[float]
[[deprecations-8.11.0]]
=== Deprecations

The following functionality is deprecated in 8.11.0, and will be removed in 9.0.0.
Deprecated functionality does not have an immediate impact on your application, but we strongly recommend
you make the necessary updates after you upgrade to 8.11.0.

[discrete]
[[deprecation-164651]]
.Updates to move from doc_root.vulnerability.package -> doc_root.package (ECS).
[%collapsible]
====
*Details* +
This updates all instances of vulnerability.package to the ECS standard package fieldset. For more information, refer to ({kibana-pull}164651[#164651]).
====
      
[float]
[[features-8.11.0]]
=== Features
{kib} 8.11.0 adds the following new and notable features.

> Alerting::

Reviewer: Yes we should mention the addition of ES|QL support for the Elasticsearch Query rule. cc @lcawl

> Alerting::

Author: Can I check with you folks, is this the best [PR](https://github.com/elastic/kibana/pull/165973) to link out to?

* Elasticsearch query rule can select multiple group-by terms ({kibana-pull}166146[#166146]).
* Adds a Log tab to the Observability Rules page ({kibana-pull}165115[#165115]).
APM::
* Adds bulk action to untrack selected alerts ({kibana-pull}167579[#167579]).
* Introduce custom dashboards tab in service overview ({kibana-pull}166789[#166789]).
* Adds service profiling Top 10 Functions ({kibana-pull}166226[#166226]).
* Adds service profiling flamegraph ({kibana-pull}165360[#165360]).
Cases::
* Adds custom fields in Cases ({kibana-pull}167016[#167016]).
Dashboard::

> * Copy panel refactor ({kibana-pull}166991[#166991]).

Reviewer 2: Maybe this title should be improved here and not mention the word refactor but what actually we are offering cc @ThomThomson

Reviewer 3: Definitely. For this one, we can say:

`Redesigned the "copy panel to Dashboard" process with a new modal and the ability to retain panel dimensions`

> * Make links panel available under technical preview ({kibana-pull}166896[#166896]).

Reviewer 3: I'm not sure what to put here, but I would consider this a highlight. If we wanted a short blurb just for the release notes it could be something like:

```suggestion
* Adds a new "Links" panel in technical preview. The Links panel allows authors to link multiple Dashboard together with fast navigation that retains users' context. ({kibana-pull}166896[#166896]).
```

> * Store view mode in local storage ({kibana-pull}166523[#166523]).

Reviewer 3: We could add more detail here. Maybe something like:

```suggestion
* Moves Dashboard view mode to local storage. This means all Dashboards will now open in the last-used view mode by default. ({kibana-pull}166523[#166523]).
```

* Adds a read only state for Managed Dashboards ({kibana-pull}166204[#166204]).
Discover::
* Adds resize support to the Discover field list sidebar ({kibana-pull}167066[#167066]).
Elastic Security::
For the Elastic Security 8.11.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].
Enterprise Search::
For the Elastic Enterprise Search 8.11.0 release information, refer to {enterprise-search-ref}/changelog.html[_Elastic Enterprise Search Documentation Release notes_].
Fleet::
* Set env variable `ELASTIC_NETINFO:false` in {kib} ({kibana-pull}166156[#166156]).
* Added restart upgrade action ({kibana-pull}166154[#166154]).
* Adds ability to set a proxy for agent binary source ({kibana-pull}164168[#164168]).
* Adds ability to set a proxy for agent download source ({kibana-pull}164078[#164078]).
Lens & Visualizations::

> * Adds color mapping for categorical dimensions in *Lens* available under technical preview, and enabled on new visualizations by default ({kibana-pull}162389[#162389]).

Reviewer 2: We decided to disable it on new visualizations for now. Let's change the text here as:
```
Adds color mapping for categorical dimensions in *Lens* available under technical preview.
```

* Inline editing of **Lens** panels on a dashboard or canvas ({kibana-pull}166169[#166169]).
* Individual annotation editing from library ({kibana-pull}163346[#163346]).
Logs::
* Convert log explorer profile into standalone app available under technical preview ({kibana-pull}164493[#164493]).
Machine Learning::
* Adds support for the ELSER v2 download in the Trained Models UI ({kibana-pull}167407[#167407]).
* Adds data drift detection workflow from Trained Models to Data comparison view ({kibana-pull}162853[#162853]).

> Management::

Reviewer 4: Can we also add this under Management?

```
* Support for viewing and editing data retention per data stream in Index Management. This feature is available under technical preview. ({kibana-pull}167006[#167006]).
```

* Index details can now be viewed on a new index details page in Index Management ({kibana-pull}165705[#165705]).

> * Adds support for enriched policies ({kibana-pull}164080[#164080]).

Reviewer 4: ```suggestion
* Support for managing, executing, and deleting enrich policies in Index Management. ({kibana-pull}164080[#164080]).
```

Platform::
* ES|QL, a new query language, is available under technical preview in Discover and Dashboards ({kibana-pull}146971[#146971]).
Querying & Filtering::
* Saved queries can now be shared between multiple spaces ({kibana-pull}163436[#163436]).
Uptime::

> * Discover document viewer has been added to the summary table ({kibana-pull}163926[#163926]).

Reviewer 5: The original PR was labeled `[Synthetics] Summary doc viewer in pings list`, I'd consider that it's the Discover document viewer as an implementation detail

Reviewer 5: it became the UnifiedDocViewer btw in the meantime, but also this is internal naming, and I don't think relevant to release notes

Author: Ah ok so the users don't really need to know it's the Discover document viewer. Let me remove the detail.

* Mark disabled alerts as Untracked in both Stack Management and o11y ({kibana-pull}164788[#164788]).
* Adds time range to event metadata API ({kibana-pull}167132[#167132]).
* New settings to control CO2 calculation ({kibana-pull}166637[#166637]).
* Adds permissions for "input-only" package ({kibana-pull}166234[#166234]).
* Adds selecting the consumer based on the authorized consumers when a user is creating an ES Query threshold rule ({kibana-pull}166032[#166032]).
* Migrate Ace based `EuiCodeEditor` to Monaco based code editor ({kibana-pull}165951[#165951]).
* Mobile UI crash widget added ({kibana-pull}163527[#163527]).
Cases::
* Show a warning message to inform user that navigating after the 10Kth case is not possible ({kibana-pull}164323[#164323]).
Dashboard::
* Focus on a single panel while disabling all other panels ({kibana-pull}165417[#165417]).
* Adds filter details to panel settings ({kibana-pull}162913[#162913]).
Discover::
* Redesign for the grid, panels and sidebar ({kibana-pull}165866[#165866]).
* Set data table row height to auto-fit by default ({kibana-pull}164218[#164218]).
* Allow fetching more documents on Discover page ({kibana-pull}163784[#163784]).

> * Fixes unified search for long field inputs ({kibana-pull}166024[#166024]).

Author: Got it

Elastic Security::
For the Elastic Security 8.11.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].
Enterprise Search::
For the Elastic Enterprise Search 8.11.0 release information, refer to {enterprise-search-ref}/changelog.html[_Elastic Enterprise Search Documentation Release notes_].
Fleet::
* Adds sidebar navigation showing headings extracted from the readme ({kibana-pull}167216[#167216]).
Lens & Visualizations::
* Other bucket defaults to false for top values greater than equal 1000 in *Lens* ({kibana-pull}167141[#167141]).
* Adds support for decimals in percentiles in *Lens* ({kibana-pull}165703[#165703]).

> * Support date fields in options list ({kibana-pull}164362[#164362]).

Reviewer 2: Are we adding the controls enhancements in the Lens section? I think we should move it to dashboard section and specify that this is about controls.

Machine Learning::
* Update ELSER version for Elastic Assistant ({kibana-pull}167522[#167522]).
* Retain `created_by` setting when exporting anomaly detection jobs ({kibana-pull}167319[#167319]).
* Improved wording of awaiting ML nodes messages ({kibana-pull}167306[#167306]).
* Adding `created_by` job property for the advanced wizard ({kibana-pull}167021[#167021]).
* Trained model testing: only show indices with supported fields ({kibana-pull}166490[#166490]).
* Alerts as data integration for Anomaly Detection rule type ({kibana-pull}166349[#166349]).

> * Data Frame Analytics Trained models: add ability to reindex after pipeline creation success ({kibana-pull}166312[#166312]).

Reviewer 6: Nit - `success` not needed at the end of this description

> * Update ELSER version for Elastic Assistant ({kibana-pull}167522[#167522]).
> * Retain `created_by` setting when exporting anomaly detection jobs ({kibana-pull}167319[#167319]).
> * Improved wording of awaiting ML nodes messages ({kibana-pull}167306[#167306]).
> * Adding `created_by` job property for the advanced wizard ({kibana-pull}167021[#167021]).
> * Trained model testing: only show indices with supported fields ({kibana-pull}166490[#166490]).
> * Alerts as data integration for Anomaly Detection rule type ({kibana-pull}166349[#166349]).
> * Data Frame Analytics Trained models: add ability to reindex after pipeline creation success ({kibana-pull}166312[#166312]).

Reviewer 7: ```suggestion
* Updates ELSER version for Elastic Assistant ({kibana-pull}167522[#167522]).
* Retains `created_by` setting when exporting anomaly detection jobs ({kibana-pull}167319[#167319]).
* Improves wording of awaiting ML nodes messages ({kibana-pull}167306[#167306]).
* Adds `created_by` job property for the advanced wizard ({kibana-pull}167021[#167021]).
* Trained model testing: only show indices with supported fields ({kibana-pull}166490[#166490]).
* Alerts as data integration for Anomaly Detection rule type ({kibana-pull}166349[#166349]).
* Data Frame Analytics Trained models: adds ability to reindex after pipeline creation ({kibana-pull}166312[#166312]).
```

* Improvements to UX of adding ML embeddables to a dashboard ({kibana-pull}165714[#165714]).

> * AIOps: Support text fields in log rate analysis ({kibana-pull}165124[#165124]).

Reviewer 7: ```suggestion
* AIOps: Supports text fields in log rate analysis ({kibana-pull}165124[#165124]).
```

* Data Frame Analytics creation wizard: adds ability to add custom URLs to jobs ({kibana-pull}164520[#164520]).
Management::

> * Adds Create a data view button to index or saved search selector in ML pages and Transforms management ({kibana-pull}166668[#166668]).

Reviewer 6: This PR applies to both the Management section (transforms) and ML. Can it be added to the Machine Learning section too?

> * Improve loading behavior of Transform ({kibana-pull}166320[#166320]).

Reviewer 6: Could the extra detail from the title of the PR be added here, something like

```
Improve loading behavior of Transforms list if stats request is slow or is not available
```

to differentiate it from the related enhancement `#164296` listed separately?

> * Clusters tab added under Inspector ({kibana-pull}166025[#166025]).

Reviewer 3: This line item is under management but should probably be moved into an `Inspector` header.

> * Index details can now be viewed on a new index details page in Index Management ({kibana-pull}165705[#165705]).

Reviewer 4: This is already mentioned under features - should it be removed here?

Author: Yes, thank you for catching that!

* Adds support for PATCH requests in Console ({kibana-pull}165634[#165634]).
* Improves autocomplete to suggest knn in search query ({kibana-pull}165531[#165531]).
* Improves display for long descriptions in Transforms ({kibana-pull}165149[#165149]).

> * Improve transform list reloading behavior in Transforms ({kibana-pull}164296[#164296]).

Reviewer 6: Nit `in Transforms` not needed at the end of this sentence.

Observability::
* ES|QL query generation ({kibana-pull}166041[#166041]).

> Presentation::

Reviewer 3: If you put both line items relating to the inspector in their own heading, you could rename this one to `Maps` potentially.

> * Open incomplete response warning in Inspector ({kibana-pull}167205[#167205]).

Reviewer 3: This could also be moved under an `inspector` header.

* Fixes set up process ({kibana-pull}167067[#167067]).
Dashboard::
* Generate new panel IDs on Dashboard clone ({kibana-pull}166299[#166299]).
Elastic Security::
For the Elastic Security 8.11.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].
Enterprise Search::
For the Elastic Enterprise Search 8.11.0 release information, refer to {enterprise-search-ref}/changelog.html[_Elastic Enterprise Search Documentation Release notes_].
Fleet::
* Vastly improve performance of Fleet final pipeline's date formatting logic for `event.ingested` ({kibana-pull}167318[#167318]).
Lens & Visualizations::
* Fixes heatmap color assignment on single value scenario in *Lens* ({kibana-pull}167995[#167995]).
* Fixes mosaic with 2 axis coloring in *Lens* ({kibana-pull}167035[#167035]).
* Show icons/titles instead of previews in suggestions panel in *Lens* ({kibana-pull}166808[#166808]).
* Consider root level filters buckets correctly when building other terms bucket ({kibana-pull}165656[#165656]).
* Prevent user to use decimals for custom Percentile rank function in Top values in *Lens* ({kibana-pull}165616[#165616]).

> * Fixes settings tabs when in dark mode ({kibana-pull}165614[#165614]).

Reviewer 2: Can we mention that this is for the Graph application somewhere in the title?

* Fixes Visualize List search and CRUD operations via content management ({kibana-pull}165485[#165485]).
Logs::
* Use correct ML API to query blocking tasks ({kibana-pull}167779[#167779]).
Machine Learning::

> * AIOps: Fixing log pattern analysis sparkles and chart ({kibana-pull}168337[#168337]).

Reviewer 6: `Sparklines` is the term to use here

Author: Thanks for catching this
